src/DistIL/AsmIO/ResolvingUtils.cs

```diff
@@ -0,0 +1,108 @@
+﻿namespace DistIL.AsmIO;
+
+using System.Collections.Concurrent;
+
+public static class ResolvingUtils
+{
+    private static readonly ConcurrentDictionary<string, MethodDesc> FunctionCache = new();
+    private static readonly ConcurrentDictionary<string, TypeDefOrSpec> TypeCache = new();
```

Comment:
Caching at static level can cause memory leaks because it will prevent the entities from being GC-ed, I'd think it would be better to instance these dictionaries inside ModuleResolver instead